ompi/mpi/man/man3/MPI_Gatherv.md

```diff
@@ -0,0 +1,379 @@
+
+MPI_GATHERV(SENDBUF, SENDCOUNT, SENDTYPE, RECVBUF, RECVCOUNTS,
+    	DISPLS, RECVTYPE, ROOT, COMM, IERROR)
+    <type>	SENDBUF(*), RECVBUF(*)
+    INTEGER	SENDCOUNT, SENDTYPE, RECVCOUNTS(*), DISPLS(*)
+    INTEGER	RECVTYPE, ROOT, COMM, IERROR
+
+MPI_IGATHERV(SENDBUF, SENDCOUNT, SENDTYPE, RECVBUF, RECVCOUNTS,
+    	DISPLS, RECVTYPE, ROOT, COMM, REQUEST, IERROR)
+    <type>	SENDBUF(*), RECVBUF(*)
+    INTEGER	SENDCOUNT, SENDTYPE, RECVCOUNTS(*), DISPLS(*)
+    INTEGER	RECVTYPE, ROOT, COMM, REQUEST, IERROR
+```
+
+## Fortran 2008 Syntax
+
+```fortran
+USE mpi_f08
+
+MPI_Gatherv(sendbuf, sendcount, sendtype, recvbuf, recvcounts, displs,
+    	recvtype, root, comm, ierror)
+    TYPE(*), DIMENSION(..), INTENT(IN) :: sendbuf
+    TYPE(*), DIMENSION(..) :: recvbuf
+    INTEGER, INTENT(IN) :: sendcount, recvcounts(*), displs(*), root
+    TYPE(MPI_Datatype), INTENT(IN) :: sendtype, recvtype
+    TYPE(MPI_Comm), INTENT(IN) :: comm
+    INTEGER, OPTIONAL, INTENT(OUT) :: ierror
+
+MPI_Igatherv(sendbuf, sendcount, sendtype, recvbuf, recvcounts, displs,
+    	recvtype, root, comm, request, ierror)
+    TYPE(*), DIMENSION(..), INTENT(IN), ASYNCHRONOUS :: sendbuf
+    TYPE(*), DIMENSION(..), ASYNCHRONOUS :: recvbuf
+    INTEGER, INTENT(IN) :: sendcount, root
+    INTEGER, INTENT(IN), ASYNCHRONOUS :: recvcounts(*), displs(*)
+    TYPE(MPI_Datatype), INTENT(IN) :: sendtype, recvtype
+    TYPE(MPI_Comm), INTENT(IN) :: comm
+    TYPE(MPI_Request), INTENT(OUT) :: request
+    INTEGER, OPTIONAL, INTENT(OUT) :: ierror
+```
+
+# Input Parameters
+
+* `sendbuf` : Starting address of send buffer (choice).
+* `sendcount` : Number of elements in send buffer (integer).
+* `sendtype` : Datatype of send buffer elements (handle).
+* `recvcounts` : Integer array (of length group size) containing the number of
+elements that are received from each process (significant only at
+root).
+* `displs` : Integer array (of length group size). Entry i specifies the
+displacement relative to recvbuf at which to place the incoming data
+from process i (significant only at root).
+* `recvtype` : Datatype of recv buffer elements (significant only at root)
+(handle).
+* `root` : Rank of receiving process (integer).
+* `comm` : Communicator (handle).
+
+
+
+# Output Parameters
+
+* `recvbuf` : Address of receive buffer (choice, significant only at root).
+* `request` : Request (handle, non-blocking only).
+* `IERROR` : Fortran only: Error status (integer).
+
+# Description
+
+
+`MPI_Gatherv` extends the functionality of `MPI_Gather` by allowing a
+varying count of data from each process, since `recvcounts` is now an
+array. It also allows more flexibility as to where the data is placed on
+the root, by providing the new argument, `displs`.
+
+The outcome is as if each process, including the root process, sends a
+message to the root,
+
+```c
+MPI_Send(sendbuf, sendcount, sendtype, root, ...)
+```
+
+and the root executes n receives,
+
+```c
+MPI_Recv(recvbuf + disp[i] * extent(recvtype), \
```

Comment:
```suggestion
MPI_Recv(recvbuf + disp[i] * extent(recvtype),
```

Reply:
Sorry for the additional \! Just deleted it.